docs/leases-and-lease-lifecycle.md

```diff
@@ -0,0 +1,166 @@
+More information, including schema, is provided at [KCL LeaseTable][kcl-leasetable].
+
+## Lease Assignment
+
+The "life" of a lease closely mirrors the duration that a shard is being processed.
+
+![Activity diagram of KCL shard-to-lease assignments.
+shard-0 (lease-0) is unmodified.
+shard-1 (lease-1) is split into shard-4 (lease-4) and shard-5 (lease-5).
+shard-2 (lease-2) and shard-3 (lease-3) are merged into shard-6 (lease-6).
+](images/leases-and-mutations.png)
+
+Specifically, leases are unique to the shard and are not recycled for stream mutations (i.e., split, merge).
+Shards created by stream mutations will generate a new lease.
+
+It should be noted that the number of tracked leases may exceed the number of shards.
+Per the diagram (above), this can occur when there are stream mutations propagating through KCL.
+For example, a 10-shard stream that is split on every shard may temporarily have up-to 30 leases: 10 original + 20 children.
+
+N.B. Leases are uniquely identified by their `leaseKey` which looks vastly different than `lease_X`.
+For details on the `leaseKey` format, please see [KCL LeaseTable][kcl-leasetable].
+
+## Lease Lifecycle
+
+Leases follow a relatively simple, progressive state machine:
+`DISCOVERY -> CREATION -> PROCESSING -> SHARD_END -> DELETION`
+
+Excluding `SHARD_END`, these phases are illustrative of KCL logic and are not explicitly codified.
+
+1. `DISCOVERY`: KCL [shard-syncing](#lease-syncing) identifies new shards.
+Discovered shards may result from:
+   * First time starting KCL with an empty lease table.
+   * Stream mutations (i.e., split, merge) that create child shards.
+   * In multi-stream mode, dynamic discovery of a new stream.
+1. `CREATION`: Leases are created 1:1 for each discovered shard, and initialized at the configured initial position.
+   * Leases are only created if they are eligible for processing.
+   For example, child shards will not have leases created until its parent(s) have reached `SHARD_END`.
+1. `PROCESSING`: Leases are processed, and continually updated with new checkpoints.
+   * In general, leases spend the majority of their life in this state.
+1. `SHARD_END`: The associated shard is `SHARD_END` and all records have been processed by KCL for the shard.
+1. `DELETION`: Since there are no more records to process, KCL will delete the lease from the lease table.
+   * Deletion will only be triggered after all parents of a child shard have converged to `SHARD_END`.
+   Convergence is required to preserve ordering of records between parent-child relationships.
+   * Deletion is configurable yet recommended to minimize I/O of lease table scans.
+
+### Lease Syncing
+
+Lease syncing is a complex responsibility owned by the "leader" host in a KCL application.
+By invoking the [ListShards API][list-shards], KCL will identify the shards for the configured stream(s).
+This process is scheduled at a
```

Comment:
You have a section on this below, but I would recommend to add it here so it doesnt seem like ListShards happens periodically but only when the auditor determines leases are missing, othterwise childshard leases are created when parent reaches SHARD_END, or something along those lines ?

Reply:
To avoid readers incorrectly assuming that `ListShards` is invoked _every_ interval, rewording (per offline consensus):
> This process is scheduled at a configurable interval so KCL can self-identify new shards introduced via stream mutations.

... to:
> This process is scheduled at a configurable interval so KCL can decide whether it should query for potentially-new shards.